python/Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py

```diff
@@ -0,0 +1,229 @@
+from __future__ import absolute_import
+
+from ..GangaUnitTest import GangaUnitTest
+
+from os import path, stat
+
+import time
+
+from .utilFunctions import getJobsPath, getXMLDir, getXMLFile, getSJXMLFile, getSJXMLIndex, getIndexFile
+
+testStr = "testFooString"
+testArgs = [[1],[2],[3],[4],[5]]
+
+class TestSJXMLGenAndLoad(GangaUnitTest):
+
+    def setUp(self):
+        """Make sure that the Job object isn't destroyed between tests"""
+        super(TestSJXMLGenAndLoad, self).setUp()
+        from Ganga.Utility.Config import setConfigOption
+        setConfigOption('TestingFramework', 'AutoCleanup', 'False')
+
+    def test_a_JobConstruction(self):
+        """ First construct the Job object (singular)"""
+        from Ganga.Utility.Config import getConfig
+        self.assertFalse(getConfig('TestingFramework')['AutoCleanup'])
+
+        from Ganga.GPI import Job, jobs, ArgSplitter
+        j=Job(splitter=ArgSplitter(args=testArgs))
+        self.assertEqual(len(jobs), 1) # Don't really gain anything from assertEqual...
+
+    def test_b_JobXMLExists(self):
+        """ Second get the job and check that getting it via jobs doesn't cause it to be loaded"""
+        from Ganga.GPI import jobs
+
+        assert len(jobs) == 1
+
+        print("len: %s" % str(len(jobs)))
```

Comment:
It does make the code noisier than necessary though. It's much cleaner to do without.